README.md

```diff
@@ -27,70 +31,103 @@ Make sure you get an API key and register a custom domain
 In `configure.swift`:
 
 ```swift
-let mailgun = Mailgun(apiKey: "<api key>", domain: "mg.example.com", region: .eu)
-services.register(mailgun, as: Mailgun.self)
+import Mailgun
+
+// Called before your application initializes.
+func configure(_ app: Application) throws {
+    /// case 1
+    /// put into your environment variables the following keys:
+    /// MAILGUN_API_KEY=...
+    app.mailgunConfiguration = .environment
+
+    /// case 2
+    /// manually
+    app.mailgunConfiguration = .init(apiKey: "<api key>")
+} 
 ```
 
 > Note: If your private api key begins with `key-`, be sure to include it
 
+### Declare all your domains
+
+```swift
+extension MailgunDomain {
+    static var myApp1: MailgunDomain { .init("mg.myapp1.com", .us) }
+    static var myApp2: MailgunDomain { .init("mg.myapp2.com", .eu) }
+    static var myApp3: MailgunDomain { .init("mg.myapp3.com", .us) }
+    static var myApp4: MailgunDomain { .init("mg.myapp4.com", .eu) }
+}
+```
+
 ### Use
 
+> Note: you could call `.mailgun(.myApp1)` from both `Application` and `Request`
+
 In `routes.swift`:
 
 #### Without attachments
 
 ```swift
-router.post("mail") { (req) -> Future<Response> in
-    let message = Mailgun.Message(
-        from: "postmaster@example.com",
-        to: "example@gmail.com",
-        subject: "Newsletter",
-        text: "This is a newsletter",
-        html: "<h1>This is a newsletter</h1>"
-    )
-
-    let mailgun = try req.make(Mailgun.self)
-    return try mailgun.send(message, on: req)
+import Mailgun
+
+func routes(_ app: Application) throws {
+    app.post("mail") { req -> EventLoopFuture<ClientResponse> in
+        let message = MailgunMessage(
+            from: "postmaster@example.com",
+            to: "example@gmail.com",
+            subject: "Newsletter",
+            text: "This is a newsletter",
+            html: "<h1>This is a newsletter</h1>"
+        )
+        return req.mailgun(.myApp1).send(message)
```

Comment:
It would be cool if you could do something like:

```swift
app.mailgun.defaultDomain = .myApp1
```

Then request handlers could easily use the default:

```swift
req.mailgun().send(...)
```

Or you could opt in to a different domain:

```swift
req.mailgun(.myApp1).send(...)
```

Reply:
Done ✅